tests/wpunit/UserObjectMutationsTest.php

```diff
@@ -969,4 +973,24 @@ public function runSendPasswordResetEmailSentTest( $mutation_arg ) {
 		return $email_sent;
 	}
 
+	public function testDisablePasswordChangedEmailsOnRegistration() {
+		$was_password_change_email_sent = false;
+
+		// If this filter is run, we know the "Password Changed" email is being sent.
+		add_filter( 'password_change_email', function( $pass_change_email ) use ( &$was_password_change_email_sent ) {
+			$was_password_change_email_sent = true;
+			return $pass_change_email;
+		} );
+
+		$this->registerUserMutation( [
+			'username' => 'password-changed-email-test-user',
+			'email'    => 'password-changed-email-test-user@example.com',
+			'password' => 'password-changed-email-test-user-password'
+		] );
+
+		/**
+		 * Assert that the "Password Changed" email was not sent when the user was registered.
+		 */
+		$this->assertFalse( $was_password_change_email_sent );
```

Comment:
I think this will always be false? The `$was_password_change_email_sent` variable in the filter callback is a variable scoped to that callback function, but this assertion is asserting the variable outside of that callback which is set to `false` and never changed.

Reply:
Is there an action that fires when an email is sent? we might be able to assert whether `did_action( 'email_sent_or_whatever' );` is true / false?

Reply:
@jasonbahl The variable inside of the callback is not scoped to that callback function, since I'm passing it in by reference with `use ( &$was_password_change_email_sent )`. So when the `$was_password_change_email_sent = true;` line runs, it updates the value of the original `$was_password_change_email_sent` variable in the outer scope to `true`.

If you comment out our new `add_filter()` line in `UserRegister.php`, then fire off a `registerUser` mutation, `$was_password_change_email_sent` will get set to `true` and the email will be sent. Alternatively, if you leave our new `add_filter()` line in `UserRegister.php`, and fire off the mutation, `$was_password_change_email_sent` will NOT get set to `true`, and the email will not be sent.

According to this trac ticket (https://core.trac.wordpress.org/ticket/35357), you can't use `did_action()` to determine whether a filter has been applied. I was thinking along those same lines, too. And there isn't a `do_action()` hook anywhere in `wp_update_user()` or `wp_mail()` that we can use.

The only other filter hook we could use would be `wp_mail`, here: https://core.trac.wordpress.org/browser/tags/5.4/src/wp-includes/pluggable.php#L181. We would need to add a filter to it and check whether the email header contains the string "Password Changed". I think my implementation is cleaner though. `wp_mail` fires for every email that gets sent, which is why we'd need to check the header text, whereas the `password_change_email` filter I'm using only fires when the changed password email is about to be sent.

Reply:
@kellenmace I overlooked `&$was_password_change_email_sent` when reviewing. 

I walked through this locally, and you've done it properly. 

Good stuff! And thanks for the great follow up response!

Reply:
@jasonbahl Sweet! Thanks for getting this one merged in. 🙌🏼